libcxx/include/__type_traits/decay.h

```diff
@@ -25,7 +25,7 @@
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
-#if __has_builtin(__decay)
+#if __has_builtin(__decay) && !defined(_LIBCPP_COMPILER_GCC)
```

Comment:
How is the GCC-provided builtin incompatible with Clang's?

Reply:
GCC doesn't mangle the builtin.

Reply:
Ah, so do we need to do something like

```
template <class _Tp>
struct __decay_impl {
  using type = __builtin_decay_t(_Tp);
};

template <class _Tp>
using __decay_t = typename __decay_impl<_Tp>::type;
```

?

Reply:
[similar issue](https://github.com/llvm/llvm-project/issues/91831)
I use the same way to fix this issue.

Reply:
Yes, something like that. I'd probably just use `decay` as the base (i.e. `using __decay_t = decay<T>::type`) for GCC.